src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala

```diff
@@ -56,6 +59,36 @@ class TasksResource @Inject() (
     "\t"
   ))
 
+  @DELETE
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  @Consumes(Array(MediaType.APPLICATION_JSON))
+  @Timed
+  def killTasks(
+    @QueryParam("scale")@DefaultValue("false") scale: Boolean,
+    body: Array[Byte]): Response = {
+    val taskIds = Json.parse(body).as[Seq[String]]
+
+    val groupedTasks = taskIds.flatMap { taskId =>
+      val appId = try {
+        taskIdUtil.appId(taskId)
```

Comment:
True, but should be done in a separate PR, because it requires lots of changes in places unrelated to this PR.